tests/baselines/reference/arityAndOrderCompatibility01.errors.txt

```diff
@@ -68,6 +69,8 @@ tests/cases/conformance/types/tuple/arityAndOrderCompatibility01.ts(30,5): error
     var [g, h, i] = z;
         ~~~~~~~~~
 !!! error TS2461: Type '{ 0: string; 1: number; }' is not an array type.
+               ~
+!!! error TS2460: Type '{ 0: string; 1: number; }' has no property '2'.

```

Comment:
Well, it's true that it has no property '2'. We weren't getting the error before because checkIteratedTypeOrElementType returns unknownType, and isTypeAny(unknownType) returns true.

Would you prefer that I put the check back, but only check the unknown type?